File: app/src/main/java/com/example/getfood/ui/orderdetail/OrderDetailPresenter.java
package com.example.getfood.ui.orderdetail;

import com.example.canteen_app_models.models.FullOrder;
import com.example.firebase_api_library.listeners.DBValueEventListener;
import com.example.getfood.ui.base.BasePresenter;

public class OrderDetailPresenter<V extends OrderDetailMvpView> extends BasePresenter<V> implements OrderDetailMvpPresenter<V> {

    public OrderDetailPresenter() {
    }


    @Override
    public void fetchOrderDetails(FullOrder fullOrder) {
        apiManager.orderDetailListener(fullOrder.getOrderId(), new DBValueEventListener<FullOrder>() {
            @Override
            public void onDataChange(FullOrder data) {
                getMvpView().bindOrderDetailAdapter(data);
            }

            @Override
            public void onCancelled(Error error) {
                getMvpView().onDatabaseError(error);
            }
        });
    }

    @Override
    public void setRatingValueForOrderItem(float rating, int position, String orderId) {
        apiManager.setRatingValueForOrderItem(rating, position, orderId);
    }
}

File: app/src/main/java/com/example/getfood/ui/orderdetail/OrderDetailPresenter.kt
package com.example.getfood.ui.orderdetail

import com.example.canteen_app_models.models.FullOrder
import com.example.firebase_api_library.listeners.DBValueEventListener
import com.example.getfood.ui.base.BasePresenter

class OrderDetailPresenter<V : OrderDetailMvpView?> : BasePresenter<V>(), OrderDetailMvpPresenter<V> {
    override fun fetchOrderDetails(fullOrder: FullOrder) {
        apiManager!!.orderDetailListener(fullOrder.orderId!!, object : DBValueEventListener<FullOrder?> {
            override fun onDataChange(data: FullOrder?) {
                data?.let {
                    mvpView?.bindOrderDetailAdapter(data)
                }
            }

            override fun onCancelled(error: Error?) {
                mvpView?.onDatabaseError(error)
            }
        })
    }

    override fun setRatingValueForOrderItem(rating: Float, position: Int, order: String?) {
        apiManager!!.setRatingValueForOrderItem(rating, position, order)
    }
}